src/platform_impl/macos/app_state.rs

```diff
@@ -98,25 +97,25 @@ impl<T> Debug for EventLoopHandler<T> {
 impl<T> EventHandler for EventLoopHandler<T> {
     fn handle_nonuser_event(&mut self, event: Event<'_, Never>, control_flow: &mut ControlFlow) {
         self.with_callback(|this, mut callback| {
-            (callback)(event.userify(), &this.window_target, control_flow);
-            this.will_exit |= *control_flow == ControlFlow::Exit;
-            if this.will_exit {
-                *control_flow = ControlFlow::Exit;
+            if let ControlFlow::ExitWithCode(code) = *control_flow {
+                let dummy = &mut ControlFlow::ExitWithCode(code);
```

Comment:
Nit: Do you need the intermediate "dummy" value here?

Reply:
Not necessarily, no. But putting it inline would cause `rustfmt` to split the callback into multiple lines, which looks even worse imo.

Reply:
Ah, fair. Could we call it `sticky` then?

Reply:
It isn't sticky at all since it only persists for that one callback, but sure, why not.

Reply:
Nah, true, `dummy` is fine